estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,77 @@
+from odoo import api, exceptions, fields, models
+
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = "An amount a potential buyer offers to the seller"
+
+    price = fields.Float(string="Price")
+    validity = fields.Integer(string="Validity", default=7)
+    status = fields.Selection(
+        string="Status",
+        copy=False,
+        selection=[
+            ('accepted', "Accepted"),
+            ('refused', "Refused")
+        ]
+    )
+
+    # relational fields
+    partner_id = fields.Many2one(comodel_name='res.partner', required=True)
+    property_id = fields.Many2one(comodel_name='estate.property', required=True)
+
+    # computed field
+    date_deadline = fields.Date(string="Deadline", compute='_compute_deadline', inverse='_inverse_deadline')
+
+    # constrain
+    _sql_constraints = [
+		('offer_price_positive', 'CHECK (price > 0.00)', 'The offer price must be strictly positive.')
+	]
+    
+    # -------------------------------------------------------------------------
+    # COMPUTE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('validity', 'create_date')
+    def _compute_deadline(self):
+        for offer in self:
+            if offer.create_date:
+                offer.date_deadline = fields.Date.add(offer.create_date, days=offer.validity)
+            else:
+                offer.date_deadline = fields.Date.add(fields.Date.today(), days=offer.validity)
+
+    # when record saved from the estate.property the inverse function will execute
+    def _inverse_deadline(self):
+        for offer in self:
+            offer.validity = (offer.date_deadline - fields.Date.today()).days
+
+    # ------------------------------------------------------------
+    # ACTIONS
+    # ------------------------------------------------------------
+
+    def action_accept(self):
+        if self.property_id.state == 'sold':
+            raise exceptions.UserError("Property already sold.")
+        elif self.property_id.state == 'cancelled':
+            raise exceptions.UserError("Property cancelled, offers cannot be accept.")
+        elif self.status == 'accepted':
+            raise exceptions.UserError("Buyer is already accepted.")
+        else:
+            # except accepted one, other offers will be refused.
+            for property in self.property_id.offer_ids:
+                if property.id != self.id:
+                    property.status = 'refused'
```

Comment:
```suggestion
            for offer in self.property_id.offer_ids:
                if offer.id != self.id:
                    offer.status = 'refused'
```
Use proper variable names in loops, as improper naming can create confusion.